create.go

```diff
@@ -58,10 +58,15 @@ func CreateWithTemplate(db *sql.DB, dir string, tmpl *template.Template, name, m
 	}
 
 	path := filepath.Join(dir, filename)
-	if _, err := os.Stat(path); !os.IsNotExist(err) {
+	_, err := os.Stat(path)
+	if err != nil && !os.IsNotExist(err) {
```

Comment:
I suggest replacing this with the code mentioned in the documentation to [os.IsNotExist](https://pkg.go.dev/os#IsNotExist:~:text=New%20code%20should%20use%20errors.Is(err%2C%20fs.ErrNotExist).) function
`!errors.Is(err, fs.ErrNotExist)`

Reply:
Also this code could be simplidfied in fort of fast failure:
```
_, err := os.Stat(path)
if err == nil {
	return fmt.Errorf("failed to create migration file: %w", os.ErrExist)
}

if !errors.Is(err, fs.ErrNotExist){
        return fmt.Errorf("failed to create migration file: %w", err)
}
```